### PR TITLE
add check if document exists

### PR DESCRIPTION
**Pull request by roccomuso**
In some environment (react native) document is not defined. This can cause errors like `Can't find variable: document`.

Should be checked first

### CODE REVIEW

**Review by kripken — APPROVED** [review]
Looks good, thanks @roccomuso  - please add yourself to AUTHORS before we merge.

In the long term I think we want https://github.com/emscripten-core/emscripten/issues/12203, but meanwhile this is good.

**Comment by roccomuso** [comment]
Done

File: src/shell.js [hunk 1]
```diff
@@ -290,7 +290,7 @@ if (ENVIRONMENT_IS_SHELL) {
 if (ENVIRONMENT_IS_WEB || ENVIRONMENT_IS_WORKER) {
   if (ENVIRONMENT_IS_WORKER) { // Check worker, not web, since window could be polyfilled
     scriptDirectory = self.location.href;
-  } else if (document.currentScript) { // web
+  } else if (typeof document !== 'undefined' && document.currentScript) { // web
```
Comment (sbc100) [inline src/shell.js:293]: does it make sense for code size to write `if (document && document.currentScript)`?   Or is the semantic difference salient here?  (or does minification do that for us?)

Comment (kripken) [reply src/shell.js:293]: There is a semantic difference, yes - `if (document` will throw if `document` is not defined at all in the scope. `typeof` is the way to check for that without throwing.

**Pull request merged**